src/inputs/select-input.jsx

```diff
@@ -172,11 +184,15 @@ class SelectInput extends React.Component {
   }
 
   sortResults = (e) => {
-    const sortValue = e.target.value && e.target.value.replace(/\s/g, '') && e.target.value.replace(/\s/g, '').length
-      ? e.target.value
+    let value = e && e.target && e.target.value && e.target.value[0] === ' '
+      ? e.target.value.trim()
+      : (e && e.target && e.target.value) || null
```

Comment:
Can you explain this ternary a bit? Why the need to check all four of `e && e.target && e.target.value && e.target.value[0]` for a value?

Reply:
I'm being extra careful with the e && e.target && e.target.value check. Just careful.

The value[0] is making sure that if you started with a space, it doesn't allow you to *just* have a space.